pkg/apps/controller/deployer/deployer_controller.go

```diff
@@ -223,6 +240,8 @@ func (c *DeploymentController) handle(deployment *v1.ReplicationController, will
 		}
 
 	case deployapi.DeploymentStatusFailed:
+		c.expectations.ExpectDeletions(key, 0)
```

Comment:
Shouldn't we expect a deletion here only if the deployment is cancelled? Also does this take into account hook pods?

Reply:
The naming of such function is unfortunate. What it means here is that we have observed a final state; decrement a counter and allow for creating a new deployer to be created if needed - usually only on manual restart, but the intend is to prevent doing so in the interim